File: service/openapi/cate.go
package openapi

import (
	"context"
	"strconv"

	apiv1 "app/proto/gen/api/v1"
	"app/store"

	"google.golang.org/protobuf/types/known/emptypb"
)

var _ apiv1.CateServiceServer = (*Cate)(nil)

type Cate struct {
	apiv1.UnimplementedCateServiceServer
	store *store.Store
}

func NewCate(s *store.Store) *Cate {
	return &Cate{store: s}
}

func (c *Cate) All(ctx context.Context, _ *emptypb.Empty) (*apiv1.CateMenuResponse, error) {
	cates, err := c.store.GetAllCates(ctx)
	if err != nil {
		return nil, err
	}
	resp := apiv1.CateMenuResponse_builder{}.Build()
	for _, v := range cates {
		resp.SetList(append(resp.GetList(), apiv1.CateMenuItem_builder{Url: "/category/" + v.Domain,
			Content: v.Name + "(" + strconv.Itoa(v.Num) + ")"}.Build(),
		))
	}
	return resp, nil
}

File: ios/Models/Cate.swift
// 分类菜单相关模型

import Foundation

/// 分类菜单项（公开接口 /blog/cate/all，URL 中仅含 domain，无数字 ID）
struct CateMenuItem: Decodable {
    let url: String
    let content: String
}

/// 分类菜单响应
struct CateMenuResponse: Decodable {
    let list: [CateMenuItem]
}

/// 分类列表项（管理端接口 /blog/admin/cate/list，包含数字 ID）
struct CateItem: Decodable, Identifiable {
    let id: Int
    let name: String
    let desc: String?
    let domain: String?
    let num: Int?
}

/// 分类列表响应（管理端）
struct CateListResponse: Decodable {
    let list: [CateItem]
    let total: Int
}
